tests/common.py

```diff
@@ -74,6 +74,8 @@
 def delete_contents(pathname):
   for entry in os.listdir(pathname):
     try_delete(os.path.join(pathname, entry))
+    # TODO(sbc): Should we make try_delete have a stronger guarantee?
+    assert not os.path.exists(os.path.join(pathname, entry))
```

Comment:
I think we can use `shutil.rmtree()` here. In the test suite we should be able to delete contents as needed. lgtm with that.

`try_delete` itself I think can stay as it is, with failure to delete being ignored.)

Reply:
But I think this is actually a bugfix for try_delete which is caring about slight the wrong thing as of today.  It should be caring about "rw" for the user but its instead insisting on "w" for user, group and other.   

I'm happy to switch the rmtree here as a followup but i think we should land this because its a bugfix

Reply:
(its also not clear to me if rmtree on its own can cleanup read-only stuff, but I'm up for trying to make that switch as a followup).

Reply:
I agree this PR is a good bugfix. Sgtm to leave the other part as a followup.